### PR TITLE
use aws sdk s3 gem

### PR DESCRIPTION
**Pull request by gotun**
## Purpose
Upgrades paperclip to use the newer `aws-sdk-s3` gem.

## Approach
Changes copied over from https://github.com/thoughtbot/paperclip/pull/2481

### CODE REVIEW

File: features/step_definitions/rails_steps.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
       gem "jruby-openssl", :platform => :jruby
       gem "capybara"
       gem "gherkin"
-      gem "aws-sdk"
+      gem "aws-sdk-s3"
```
Comment (cameronlockey) [inline features/step_definitions/rails_steps.rb:21]: Nice! This pares down what we are using/importing to just S3 service since that's all we are using here?

File: spec/paperclip/storage/s3_spec.rb [hunk 2]
```diff
@@ -282,7 +282,7 @@ class << @dummy
     end
   end
 
-  context "use_accelerate_endpoint", if: aws_accelerate_available? do
+  context "use_accelerate_endpoint" do
```
Comment (cameronlockey) [inline spec/paperclip/storage/s3_spec.rb:285]: can you comment on why this change was needed?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -39,7 +39,6 @@
   config.include TestData
   config.include Reporting
   config.extend VersionHelper
-  config.extend ConditionalFilterHelper
```
Comment (cameronlockey) [inline spec/spec_helper.rb:42]: Mind just leaving a comment on why we are removing this?

File: spec/paperclip/storage/s3_spec.rb (hunk 2 above)
Comment (gotun) [reply spec/paperclip/storage/s3_spec.rb:285]: All these changes are copied over from the changes made to the official paperclip gem. See the description. This option was not available for a certain range of versions of the aws sdk gem. That's why it's conditional here. It is available for `aws-sdk-s3`

File: spec/spec_helper.rb (hunk 3 above)
Comment (gotun) [reply spec/spec_helper.rb:42]: If you expand the file below, it will all make sense.
